### PR TITLE
Add WCS information to the WorkUnit

### PR DESCRIPTION
**Pull request by jeremykubica**
Adds two forms of WCS information to the `WorkUnit` class. A global `wcs` attribute that should be set when all the images are projected into the same pixel space and a `per_image_wcs` for images that live in different pixel spaces. This information is automatically stored in and read from the FITS headers.

### CODE REVIEW

File: tests/test_work_unit.py [hunk 1]
```diff
@@ -40,11 +41,37 @@ def setUp(self):
         self.config.set("mask_bits_dict", {"A": 1, "B": 2})
         self.config.set("repeated_flag_keys", None)
 
+        # Creater a fake WCS
```
Comment (delucchi-cmu) [inline tests/test_work_unit.py:44]:
```suggestion
        # Create a fake WCS
```

File: tests/test_work_unit.py [hunk 2]
```diff
@@ -93,6 +121,9 @@ def test_save_and_load_fits(self):
                     for x in range(p1.get_dim()):
                         self.assertAlmostEqual(p1.get_value(y, x), p2.get_value(y, x))
 
+                # No per-image WCS
```
Comment (wilsonbb) [inline tests/test_work_unit.py:124]: Can we test generating a per-image WCS as well?

File: src/kbmod/work_unit.py [hunk 3]
```diff
@@ -77,7 +93,9 @@ def from_fits(cls, filename):
                 imgs.append(LayeredImage(sci, var, msk, p))
 
         im_stack = ImageStack(imgs)
-        return WorkUnit(im_stack=im_stack, config=config)
+        result = WorkUnit(im_stack=im_stack, config=config, wcs=global_wcs)
+        result.per_image_wcs = per_image_wcs
```
Comment (wilsonbb) [inline src/kbmod/work_unit.py:97]: I probably would have set `self.per_image_wcs` via an optional parameter to the `WorkUnit`'s constructor.

This seems fine but curious if there was a specific reason we chose not to do that

File: tests/test_work_unit.py (hunk 2 above)
Comment (jeremykubica) [reply tests/test_work_unit.py]: Good call. I added a mix of WCS and non-WCS to test the parsing functions.

File: src/kbmod/work_unit.py (hunk 3 above)
Comment (jeremykubica) [reply src/kbmod/work_unit.py]: I changed it to use an optional parameter.

Originally the motivation is the most `WorkUnit`s should use the single wcs. The `per_image_wcs` is only for ones that have not been reprojected yet. But I think it is clearer the way you proposed.

**Review by wilsonbb — APPROVED** [review]
LGTM!

**Pull request merged**